coolest/api/plotting.py

```diff
@@ -444,3 +447,138 @@ def plot_source(self,idx_file=0):
+    chain_objs : array
+        A list of coolest objects that have a chain file associated to them.
+    chain_dirs : array
+        A list of paths matching the coolest files in 'chain_objs'.
+    chain_names : array, optional
+        A list of labels for the coolest models in the 'chain_objs' list. Must have the same order as 'chain_objs'.
+    point_estimate_objs : array, optional
+        A list of coolest objects that will be used as point estimates.
+    point_estimate_dirs : array
+        A list of paths matching the coolest files in 'point_estimate_objs'.
+    point_estimate_names : array, optional
+        A list of labels for the models in the 'point_estimate_objs' list. Must have the same order as 'point_estimate_objs'.
+    labels : dict, optional
+        A dictionary matching the parameter_id_list entries to some human-readable labels.
+    
+
+    Returns
+    -------
+    An image
+    """
+
+    chains.print_load_details = False # Just to silence messages
+    parameter_id_set = set(parameter_id_list)
+    Npars = len(parameter_id_list)
+    
+    # Get the chain file headers from the first object in the list
+    chain_file = os.path.join(chain_dirs[0],chain_objs[0].meta["chain_file_name"])
+    
+
+    # Set the chain names
+    if chain_names is None:
+        chain_names = ["chain "+str(i) for i in range(0,len(chain_objs))]
+    
+
+    # Get the values of the point_estimates
+    point_estimates = []
+    if point_estimate_objs is not None:
+        for coolest_obj in point_estimate_objs:
+            values = []
+            for par in parameter_id_list:
+                param = coolest_obj.lensing_entities.get_parameter_from_id(par)
+                val = param.point_estimate.value
+                if val is None:
+                    values.append(None)
+                else:
+                    values.append(val)
+            point_estimates.append(values)
+
+            
+    mcsamples = []
+    for i in range(0,len(chain_objs)):
+        chain_file = os.path.join(chain_dirs[i],chain_objs[i].meta["chain_file_name"]) # Here get the chain file path for each coolest object
+
+        # Each chain file can have a different number of free parameters
+        f = open(chain_file)
+        header = f.readline()
+        f.close()
+        chain_file_headers = header.split(',')
+        chain_file_headers.pop() # Remove the last column name that is the probability weights
+        chain_file_headers_set = set(chain_file_headers)
+        
+        # Check that the given parameters are a subset of those in the chain file
+        assert parameter_id_set.issubset(chain_file_headers_set), "Not all given parameters are free parameters for model %d (not in the chain file: %s)!" % (i,chain_file)
+
+        # Set the labels for the parameters in the chain file
+        par_labels = []
+        if labels is None:
```

Comment:
What about getting the `.latex_str` attribute of the Parameter instance instead (and only if the `.latex_str` is None then you set the ID)? Also, I would check each label in the list individually, so that the user can give some of the them, and leave None when they want the default.

Reply:
Good suggestion, I will implement this asap.

Reply:
Done. I had to modify the function 'get_parameter_from_id' to return None instead of raising an exception if the parameter id does not correspond to any in the lensing entities. Is that ok?

Reply:
Yep it's fine for me